### PR TITLE
Update Http  links with Htpps 

### PR DESCRIPTION
**Pull request by abhi-glitchhg**
Updating the HTTP links with HTTPs;

### CODE REVIEW

**Review by NicolasHug — COMMENTED** [review]
Thanks for the PR @abhi-glitchhg ,

unfortunately I'm not sure we will merge it. I just picked one random link: https://host.robots.ox.ac.uk/pascal/VOC

and it doesn't load. I'm really sorry but I won't have time to check them all (maybe a script could do that for me, but I don't have the bandwidth).

I could be wrong but I think websites or users can force https from a http link, while the reverse isn't true. So I'm not sure switching all links to https adds a significant security layer: those who want it will have it already

File: torchvision/datasets/_optical_flow.py [hunk 1]
```diff
@@ -466,7 +466,7 @@ def __getitem__(self, index: int) -> Union[T1, T2]:
 def _read_flo(file_name: str) -> np.ndarray:
     """Read .flo file in Middlebury format"""
     # Code adapted from:
-    # http://stackoverflow.com/questions/28013200/reading-middlebury-flow-files-with-python-bytes-array-numpy
+    # https://stackoverflow.com/questions/28013200/reading-middlebury-flow-files-with-python-bytes-array-numpy
```
Comment (oke-aditya) [inline torchvision/datasets/_optical_flow.py:469]: While this link works fine.

File: torchvision/datasets/_optical_flow.py [hunk 2]
```diff
@@ -397,7 +397,7 @@ def _read_flow(self, file_name: str) -> np.ndarray:
 
 
 class HD1K(FlowDataset):
-    """`HD1K <http://hci-benchmark.iwr.uni-heidelberg.de/>`__ dataset for optical flow.
+    """`HD1K <https://hci-benchmark.iwr.uni-heidelberg.de/>`__ dataset for optical flow.
```
Comment (oke-aditya) [inline torchvision/datasets/_optical_flow.py:400]: There is no https link. This doesn't open.

File: torchvision/datasets/_optical_flow.py [hunk 3]
```diff
@@ -170,7 +170,7 @@ def _read_flow(self, file_name: str) -> np.ndarray:
 
 
 class KittiFlow(FlowDataset):
-    """`KITTI <http://www.cvlibs.net/datasets/kitti/eval_scene_flow.php?benchmark=flow>`__ dataset for optical flow (2015).
+    """`KITTI <https://www.cvlibs.net/datasets/kitti/eval_scene_flow.php?benchmark=flow>`__ dataset for optical flow (2015).
```
Comment (oke-aditya) [inline torchvision/datasets/_optical_flow.py:173]: This works fine

File: torchvision/datasets/_optical_flow.py [hunk 4]
```diff
@@ -83,7 +83,7 @@ def __rmul__(self, v: int) -> torch.utils.data.ConcatDataset:
 
 
 class Sintel(FlowDataset):
-    """`Sintel <http://sintel.is.tue.mpg.de/>`_ Dataset for optical flow.
+    """`Sintel <https://sintel.is.tue.mpg.de/>`_ Dataset for optical flow.
```
Comment (oke-aditya) [inline torchvision/datasets/_optical_flow.py:86]: This is one bad case. Where https leads to some other website, whereas http leads to some other.

**Review by oke-aditya — CHANGES_REQUESTED** [review]
Let me chime in my thoughts :smile: And it might clarify lot of things

HTTPs  Requires SSL and TLS connections. This needs SSL certificate to be installed on the server etc.  Which won't be true for most dataset URLs.
 
Note that when we try opening urls from browser. Modern browsers like Firefox have **Force websites to use HTTPs over HTTP** option. So even if the url is HTTP the browser will try to open it in HTTPs.

You can try that with Kitti url. 

The harder part here is say a maintainer forgets to renew his SSL certificates. The HTTPs url won't work. While the HTTP will work fine. 

HTTP is not secure as HTTPs but as such the user can take of this browsing. As such we can't keep track of HTTP vs HTTPs urls for many datasets.


Read here

https://web.dev/enable-https/
https://www.godaddy.com/garage/enable-https-server/

**Comment by NicolasHug** [comment]
Thanks for your input @oke-aditya , sounds like we're on the same page. I'll close the PR, thanks for opening the discussion @abhi-glitchhg

**Pull request closed without merging**